pkg/spdk/server.go

```diff
@@ -104,8 +105,41 @@ func (s *Server) monitoring() {
 			logrus.Info("SPDK Server: stopped monitoring replicas due to the context done")
 			done = true
 		case <-ticker.C:
-			if err := s.verify(); err != nil {
-				logrus.WithError(err).Errorf("SPDK Server: failed to verify and update replica cache, will retry later")
+			err := s.verify()
+			if err == nil {
+				break
+			}
+
+			logrus.WithError(err).Errorf("SPDK Server: failed to verify and update replica cache, will retry later")
+
+			if !jsonrpc.IsJSONRPCRespErrorBrokenPipe(err) {
+				break
+			}
+
+			running, err := util.IsSPDKTargetProcessRunning()
+			if err != nil {
+				logrus.WithError(err).Errorf("SPDK Server: failed to check spdk_tgt is running")
+				break
+			}
+			if running {
+				logrus.Info("SPDK Server: reconnecting to spdk_tgt")
+
+				cli, err := NewSPDKClient()
+				if err != nil {
+					logrus.WithError(err).Errorf("SPDK Server: failed to reconnect to spdk_tgt")
+					break
+				}
+
+				err = s.spdkClient.Close()
+				if err != nil {
+					logrus.WithError(err).Warnf("SPDK Server: failed to close spdk_tgt connection")
+				}
+				s.spdkClient = cli
+			} else {
+				err = util.StartSPDKTgtDaemon()
+				if err != nil {
+					logrus.WithError(err).Errorf("SPDK Server: failed to start spdk_tgt daemon")
+				}
 			}
```

Comment:
How about moving this code to a private method instead? like `tryEnsureSPDKTgtHealthy`

Reply:
Addressed, thank you.